extractor/extractor.cpp

```diff
@@ -572,9 +572,17 @@ void extractor::WriteNodeMapping(const std::vector<QueryNode> & internal_to_exte
 
     Saves tree into '.ramIndex' and leaves into '.fileIndex'.
  */
-void extractor::BuildRTree(const std::vector<EdgeBasedNode> &node_based_edge_list,
+void extractor::BuildRTree(std::vector<EdgeBasedNode> &&node_based_edge_list_,
                          const std::vector<QueryNode> &internal_to_external_node_map)
 {
+    std::vector<EdgeBasedNode> node_based_edge_list(std::move(node_based_edge_list_));
+
+    // Filter out all segments that are not traversible by the default mode of transportation
+    // For example this filters out ferry routes and such.
+    std::remove_if(node_based_edge_list.begin(), node_based_edge_list.end(), [](const EdgeBasedNode& segment) {
+                return segment.forward_travel_mode != TRAVEL_MODE_DEFAULT && segment.backward_travel_mode != TRAVEL_MODE_DEFAULT;
+            });
+
```

Comment:
Hmm looks like you forgot to `erase`? `remove_if` will only shuffle items satisfying the predicate to the end.